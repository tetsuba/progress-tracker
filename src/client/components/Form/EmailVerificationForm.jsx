import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import { Button, Form, Col } from 'react-bootstrap'

// HOOKS
import { useInputChange } from '../../hooks/hooks'

// COMPONENTS
import Box from '../Box/Box'

// ICONS
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCaretLeft } from '@fortawesome/free-solid-svg-icons'

export default function EmailVerificationForm(props) {
  const { handleSubmit, error } = props
  const [errorMessage] = useState(() => error && error.graphQLErrors[0].message)
  const [inputs, setInputs] = useInputChange({ email: '' })
  const options = { variables: { input: inputs } }

  return (
    <Box max={500}>
      <Col>
        <h3 className="text-danger">Could not sign you in</h3>
        <p>
          Your email address has not been confirmed. Please enter your email
          address below and a link will be sent to confirm your email address.
        </p>
        <Form
          id="EmailVerificationForm"
          onSubmit={(e) => {
            e.preventDefault()
            handleSubmit(options)
          }}
        >
          <Form.Group controlId="EmailVerificationEmail">
            <Form.Control
              required
              type="email"
              placeholder="email@address.com"
              name="email"
              onChange={setInputs}
              value={inputs.email}
              isInvalid={!!errorMessage}
            />
            <Form.Control.Feedback type="invalid">
              {errorMessage}
            </Form.Control.Feedback>
            <Form.Text className="text-muted">
              We'll never share your email with anyone else.
            </Form.Text>
          </Form.Group>
          <Button
            id="EmailVerificationSubmit"
            className="float-right"
            variant="primary"
            type="submit">
            Send
          </Button>
        </Form>
        <Link to="/">
          <FontAwesomeIcon icon={faCaretLeft} /> back
        </Link>
      </Col>
    </Box>
  )
}
