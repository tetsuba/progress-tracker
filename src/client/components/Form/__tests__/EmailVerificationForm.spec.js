import { testRenderer } from '../../../../test/testHelper'

// COMPONENTS
import EmailVerificationForm from '../EmailVerificationForm'

describe('<EmailVerificationForm>', () => {
  const props = {
    handleSubmit: jest.fn(),
    error: undefined,
  }

  it('should render ', () => {
    const wrapper = testRenderer(EmailVerificationForm, props)
    expect(wrapper).toMatchSnapshot()
  })

  describe('@Events', () => {
    describe('onSubmit', () => {
      it('should trigger handleSubmit', () => {
        const wrapper = testRenderer(EmailVerificationForm, props)

        wrapper.find('#EmailVerificationForm')
          .props()
          .onSubmit({ preventDefault: jest.fn() })

        expect(props.handleSubmit).toHaveBeenCalledTimes(1)
      })
    })

    describe('onChange', () => {
      it('should update email input', () => {
        const wrapper = testRenderer(EmailVerificationForm, props)
        const email = { name: 'email' }
        const expected = 'test@test.com'

        wrapper.find(email).simulate('change', {
          persist: jest.fn(),
          target: {
            name: 'email',
            value: 'test@test.com',
          },
        })
        expect(wrapper.find(email).prop('value')).toEqual(expected)
      })
    })
  })

  describe('@Error', () => {
    it('should display an error if error is provided', () => {
      const errorMessage = 'This is an error message'
      const error = {
        graphQLErrors: [
          {
            message: errorMessage
          }
        ]
      }
      const wrapper = testRenderer(EmailVerificationForm, {
        ...props,
        error: error,
      })
      expect(wrapper.find({type: 'invalid'}).prop('children')).toEqual(errorMessage)
    })
  })
})
