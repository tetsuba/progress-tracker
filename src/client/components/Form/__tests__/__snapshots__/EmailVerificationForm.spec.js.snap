// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailVerificationForm> should render  1`] = `
<Box
  max={500}
>
  <Col>
    <h3
      className="text-danger"
    >
      Could not sign you in
    </h3>
    <p>
      Your email address has not been confirmed. Please enter your email address below and a link will be sent to confirm your email address.
    </p>
    <Form
      id="EmailVerificationForm"
      inline={false}
      onSubmit={[Function]}
    >
      <FormGroup
        controlId="formBasicEmail"
      >
        <FormControl
          isInvalid={false}
          name="email"
          onChange={[Function]}
          placeholder="email@address.com"
          required={true}
          type="email"
          value=""
        />
        <Feedback
          type="invalid"
        />
        <FormText
          className="text-muted"
        >
          We'll never share your email with anyone else.
        </FormText>
      </FormGroup>
      <Button
        active={false}
        className="float-right"
        disabled={false}
        type="submit"
        variant="primary"
      >
        Send
      </Button>
    </Form>
    <Link
      to="/"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              192,
              512,
              Array [],
              "f0d9",
              "M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z",
            ],
            "iconName": "caret-left",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       back
    </Link>
  </Col>
</Box>
`;
